validator/validator_spawner.go

```diff
@@ -111,11 +270,29 @@ func (v *ArbitratorSpawner) Execute(
 	return mach.GetGlobalState(), nil
 }
 
+func (v *ArbitratorSpawner) Launch(entry *ValidationInput, moduleRoot common.Hash) ValidationRun {
+	atomic.AddInt32(&v.count, 1)
+	run := NewvalRun(moduleRoot)
+	v.LaunchThread(func(ctx context.Context) {
+		run.consumeResult(v.execute(ctx, entry, moduleRoot))
+		atomic.AddInt32(&v.count, -1)
```

Comment:
I'd recommend `defer atomic.AddInt32(&v.count, -1)` instead at the top of this function so it'd still decrement on panic

Reply:
Interesting. I need to learn how panic recovery works.